### PR TITLE
Zombie game logic rules

### PR DESCRIPTION
**Pull request by tamarum**
## Description

Resolves sc-story-id

Logic rules for the different health states during the game. Includes calculations for time dependent and exposure based progress.

## Testing

Varied state thresholds and # of infected humans/cats over 20 scan steps. Tested settings to trigger immune and zombie states. State transitions behave as expected.

## Notes

Be kind :)

Be sure to not only [request reviewers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) but also [assign someone to the pull request](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users) so it is clear when a review is required.

### CODE REVIEW

**Comment by tamarum** [comment]
@tylerjw @briancairl

File: scripts/game_rules.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+def is_zombie(health):
+    return State_bounds.ZOMBIE <= health
+
+
+# Determine health count
+def health_count(health,exposures_human,exposures_cat):
+    if is_zombie(health):
+        return State_bounds.ZOMBIE
+    if is_immune(health):
```
Comment (briancairl) [inline scripts/game_rules.py:58]: `elif`

File: scripts/game_rules.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        return State_bounds.IMMUNE
+    health_current = health
+    health = health_current + progress(health_current) + exposure(health_current,exposures_human,exposures_cat)
```
Comment (briancairl) [inline scripts/game_rules.py:61]: can be simplified to:
```py
health += progress(health) + exposure(health, exposures_human,exposures_cat)
```

File: scripts/game_rules.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        return State_bounds.ZOMBIE
+    return health
+
+def progress(health): #Checks health status, returns a 0 or constant value to increment h by to track disease progress
+    if is_super_healthy(health):
+        sum = health + Progress_rate.SUPER_HEALTHY
+        remainder = sum % State_bounds.HEALTHY
+        quotient = sum // State_bounds.HEALTHY
+        return  Progress_rate.SUPER_HEALTHY - remainder*quotient
+    if is_infected(health):
```
Comment (briancairl) [inline scripts/game_rules.py:72]: `elif`

**Review by briancairl — APPROVED** [review]
Solid. Just some small nits.

**Pull request merged**